src/notebook/model.ts

```diff
@@ -496,141 +317,153 @@ class NotebookModel implements INotebookModel {
     }
     cells.clear();
     this._cells = null;
```

Comment:
We should `clearSignalData(this._cells)` too. Or even better, perhaps the ObservableList should implement IDisposable, and we just call `dispose()` instead of `clear()`, and that would also clear the signal data.